cloud/openstack/os_nova_flavor.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/python
+
+# Copyright (c) 2015 Hewlett-Packard Development Company, L.P.
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_nova_flavor
+short_description: Manage OpenStack compute flavors
+extends_documentation_fragment: openstack
+version_added: "2.0"
+author: "David Shrewsbury (@Shrews)"
+description:
+   - Add or remove flavors from OpenStack.
+options:
+   state:
+     description:
+        - Indicate desired state of the resource.
+     choices: ['present', 'absent']
+     required: false
+     default: present
+   name:
+     description:
+        - Flavor name.
+     required: true
+   ram:
+     description:
+        - Amount of memory, in MB.
+     required: false (true if I(state) is 'present')
+   vcpus:
+     description:
+        - Number of virtual CPUs.
+     required: false (true if I(state) is 'present')
+   disk:
+     description:
+        - Size of local disk, in GB.
+     required: false (true if I(state) is 'present')
+   ephemeral:
+     description:
+        - Ephemeral space size, in GB.
+     required: false
+     default: 0
+   swap:
+     description:
+        - Swap space size, in MB.
+     required: false
+     default: 0
+   rxtx_factor:
+     description:
+        - RX/TX factor.
+     required: false
+     default: 1.0
+   is_public:
+     description:
+        - Make flavor accessible to the public.
+     required: false
+     default: true
+   flavorid:
+     description:
+        - ID for the flavor. This is optional as a unique UUID will be
+          assigned if a value is not specified.
+     required: false
+     default: "auto"
+requirements: ["shade"]
+'''
+
+EXAMPLES = '''
+# Create 'tiny' flavor with 1024MB of RAM, 1 virtual CPU, and 10GB of
+# local disk, and 10GB of ephemeral.
+- os_nova_flavor:
+    cloud=mycloud
+    state=present
+    name=tiny
+    ram=1024
+    vcpus=1
+    disk=10
+    ephemeral=10
+
+# Delete 'tiny' flavor
+- os_nova_flavor:
+    cloud=mycloud
+    state=absent
+    name=tiny
+'''
+
+RETURN = '''
+flavor:
+    description: Dictionary describing the flavor.
+    returned: success
```

Comment:
How do I document this correctly? There are few examples of this new documentation.  :(